### ISSUE TITLE
Re-fitting Model on Testing Set

### ISSUE THREAD

**Issue by nvk23** [GenoML/genoml2#48]
Line 50 at [discrete testing](https://github.com/GenoML/genoml2/tree/master/genoml/discrete/supervised/testing.py) and Line 59 at [continuous testing](https://github.com/GenoML/genoml2/blob/master/genoml/continuous/supervised/testing.py) re-fits tuned model on unseen test set which reports inflated results. Tuned model joblib files also aren't saved with estimators from tuned dataset.

_Discrete Testing:_
```
def plot_results(self, save=False):
        # Issue #24: RandomForestClassifier is finicky - can't recalculate moving forward like the other
        self.algo.fit(self.X_test, self.y_test)
        plot_path = self.run_prefix.joinpath('testModel_withheldSample_ROC.png')
        ground_truth = self.y_test.values
        predictions = self.algo.predict(self.X_test)
        discrete_utils.ROC(save, plot_path, ground_truth, predictions)
        discrete_utils.precision_recall_plot(save, plot_path, ground_truth, predictions)
```

_Continuous Testing:_
```
def performance_metrics(self):
        
        log_cols=["Explained_variance_score", "Mean_squared_error", "Median_absolute_error", "R2_score"]
        log_table = pd.DataFrame(columns=log_cols)
                
        self.loaded_model.fit(self.X_test, self.y_test)

        print("")
        print("#"*70)

        test_predictions = self.loaded_model.predict(self.X_test)
        test_predictions = test_predictions
        evs = explained_variance_score(self.y_test, test_predictions)
        print("Explained variance score: {:.4}".format(evs))
```
